src/services/fs/dir_stream.rs

```diff
@@ -113,15 +128,21 @@ impl BlockingObjectPage for BlockingDirPager {
             let file_type = de.file_type().map_err(parse_io_error)?;
 
             let d = if file_type.is_file() {
-                ObjectEntry::new(&path, ObjectMetadata::new(ObjectMode::FILE))
+                ObjectEntry::new(
+                    &rel_path.to_string_lossy(),
```

Comment:
We need to make sure Object uses `path/to/file`, does this call meets this requirement?

Reply:
Nope! will fix!